language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/BlockNodeContextProvider.java

```diff
@@ -137,7 +141,21 @@ public BlockNodeContextProvider(Class<T> attachmentPoint) {
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETRY.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETRY_TRANSACTION.get()));
         completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_MATCH.get()));
-        completionItems.add(new SnippetCompletionItem(context, Snippet.STMT_RETURN.get()));
+
+
+        Optional<SemanticModel> semanticModel = context.currentSemanticModel();
+        if (semanticModel.isEmpty()) {
+            return completionItems;
```

Comment:
Shall we use isPresent check and include the logic there. Otherwise we'll be ignoring the completion Items which are added below this line